cineast-core/src/main/java/org/vitrivr/cineast/core/temporal/ScoredSegment.java

```diff
@@ -0,0 +1,97 @@
+package org.vitrivr.cineast.core.temporal;
+
+import org.jetbrains.annotations.NotNull;
+import org.vitrivr.cineast.core.data.StringDoublePair;
+import org.vitrivr.cineast.core.data.entities.MediaSegmentDescriptor;
+
+/**
+ * Class to store the information of a scored segment.
+ *
+ * <p>Stores relevant information and score of a segment of a container</p>
+ *
+ * @author vGsteiger
+ * @created 11.05.2021
+ */
+public class ScoredSegment implements Comparable<ScoredSegment> {
+
+  private final String objectId;
+  private final String segmentId;
+  private final int containerId;
+  private final float segmentLength;
+  private final float startAbs;
+  private final float endAbs;
+  private double score;
+
+  /**
+   * Constructor to create a scored segment.
+   */
+  public ScoredSegment(MediaSegmentDescriptor mediaSegmentDescriptor, double score, int containerId, float segmentLength) {
+    this.objectId = mediaSegmentDescriptor.getObjectId();
+    this.segmentId = mediaSegmentDescriptor.getSegmentId();
+    this.startAbs = mediaSegmentDescriptor.getStartabs();
+    this.endAbs = mediaSegmentDescriptor.getEndabs();
+    this.containerId = containerId;
+    this.segmentLength = segmentLength;
+    this.score = score;
+  }
+
+  /**
+   * Copy constructor to create a new instance with the same values.
+   */
+  public ScoredSegment(ScoredSegment scoredSegment) {
+    this.objectId = scoredSegment.getObjectId();
+    this.segmentId = scoredSegment.getSegmentId();
+    this.score = scoredSegment.getScore();
+    this.containerId = scoredSegment.getContainerId();
+    this.segmentLength = scoredSegment.getSegmentLength();
+    this.startAbs = scoredSegment.getStartAbs();
+    this.endAbs = scoredSegment.getEndAbs();
+  }
+
+  /**
+   * Add a score to this segment, meaning this segment appeared more than once in a container.
+   */
+  public void addScore(StringDoublePair stringDoublePair) {
+    if (stringDoublePair.key.equals(this.segmentId) && stringDoublePair.value > 0) {
+      this.score += stringDoublePair.value;
+    }
+  }
+
+  public String getSegmentId() {
+    return segmentId;
+  }
+
+  public double getScore() {
+    return score;
+  }
+
+  public int getContainerId() {
+    return containerId;
+  }
+
+  public float getSegmentLength() {
+    return segmentLength;
+  }
+
+  public String getObjectId() {
+    return objectId;
+  }
+
+  public float getStartAbs() {
+    return startAbs;
+  }
+
+  public float getEndAbs() {
+    return endAbs;
+  }
+
+  @Override
+  public int compareTo(@NotNull ScoredSegment o) {
```

Comment:
I think it would be helpful to have some context here as to why first the container, and then the segmentID is compared.

Reply:
I wrote a comment in the new commit that hopefully clears up this question.